python/lsst/obs/base/defineVisits.py

```diff
@@ -624,18 +654,43 @@ def computeExposureBounds(self, exposure: DimensionRecord, *, collections: Any =
         camera, versioned = loadCamera(self.butler, exposure.dataId, collections=collections)
         if not versioned and self.config.requireVersionedCamera:
             raise LookupError(f"No versioned camera found for exposure {exposure.dataId}.")
-        if self.config.detectorId is None:
-            wcsRefs = list(self.butler.registry.queryDatasets("raw.wcs", dataId=exposure.dataId,
-                                                              collections=collections))
-            if not wcsRefs:
-                raise LookupError(f"No raw.wcs datasets found for data ID {exposure.dataId} "
-                                  f"in collections {collections}.")
-            wcsDetector = camera[wcsRefs[0].dataId["detector"]]
-            wcs = self.butler.getDirect(wcsRefs[0])
+
+        # Derive WCS from boresight information -- if available in registry
+        use_registry = True
+        try:
+            orientation = lsst.geom.Angle(exposure.sky_angle, lsst.geom.degrees)
+            radec = lsst.geom.SpherePoint(lsst.geom.Angle(exposure.tracking_ra, lsst.geom.degrees),
+                                          lsst.geom.Angle(exposure.tracking_dec, lsst.geom.degrees))
+        except AttributeError:
+            use_registry = False
+
+        if use_registry:
+            if self.config.detectorId is None:
+                detectorId = next(camera.getIdIter())
+            else:
+                detectorId = self.config.detectorId
+            wcsDetector = camera[detectorId]
+
+            # Ask the raw formatter to create the relevant WCS
+            # This allows flips to be taken into account
+            instrument = self.getInstrument(exposure.instrument)
+            rawFormatter = instrument.getRawFormatter({"detector": detectorId})
```

Comment:
I don't think there's much we can do about this now, but I'm vaguely worried that someday we'll swap in a new `Formatter` for some instrument because we start writing images out differently, and we'll have no way to connect a particular `Formatter` to a particular exposure.  I suppose if we can put enough information into cameraGeom as discussed on Slack, that ceases to be a problem in at least this case.

Reply:
I suppose it is technically possible for new data to need a different formatter to older data. In theory datastore knows exactly which formatter to use since it's stored there. We haven't formally got any public method for asking datastore though.

Reply:
This problem potentially affects `Datastore` as well - if you have a new repo and ingest old raws into it with a new stack, you could be ingested with the wrong `Formatter` if it's changed from the original.

Reply:
Yes, effectively we can't handle that. If the format changes (Say from FITS to HDF) we are going to have to require that the formatter knows how to deal with both.